python/cuml/test/experimental/test_explainer_kernel_shap.py

```diff
@@ -80,180 +68,134 @@ def experimental_test_and_log(cu_shap_values,
-                                         n_features=nfeatures,
-                                         noise=0.1, dtype=dtype)
-
-    X_train, X_test, y_train, y_test = train_test_split(
-        X, y, test_size=2)
+def test_kernel_shap_standalone(dtype, n_features, n_background, model):
+    X_train, X_test, y_train, y_test = create_synthetic_dataset(
+        n_samples=n_background + 3,
+        n_features=n_features,
+        test_size=3,
+        noise=0.1,
+        dtype=dtype
+    )
 
     mod = model(n_components=3).fit(X_train, y_train)
+    explainer, shap_values = get_shap_values(
+        model=mod.transform,
+        background_dataset=X_train,
+        explained_dataset=X_test,
+        explainer=cuKE
+    )
 
-    cu_explainer = \
-        cuml.experimental.explainer.KernelExplainer(model=mod.transform,
-                                                    data=X_train,
-                                                    is_gpu_model=True)
-
-    cu_shap_values = cu_explainer.shap_values(X_test)
-    exp_v = cu_explainer.expected_value
+    exp_v = explainer.expected_value
 
-    # we have 5 lists of shap values, each corresponding to a component since
+    # we have 3 lists of shap values, each corresponding to a component since
     # transform gives back arrays of shape (nrows x ncomponents)
     # we test that for each test row, for each component, the
     # sum of the shap values is the same as the difference between the
     # expected value for that component minus the value of the transform of
     # the row.
-    for sv_idx in range(2):
+    for sv_idx in range(3):
         # pca and tsvd transform give results back nested
-        fx = mod.transform(X_test[sv_idx].reshape(1, nfeatures))[0]
+        fx = mod.transform(X_test[sv_idx].reshape(1, n_features))[0]
 
         for comp_idx in range(3):
             assert(
                 np.sum(
-                    cu_shap_values[comp_idx][sv_idx]) - abs(
+                    shap_values[comp_idx][sv_idx]) - abs(
                         fx[comp_idx] - exp_v[comp_idx])) <= 1e-5
 
 
 @pytest.mark.parametrize("dtype", [np.float32, np.float64])
-@pytest.mark.parametrize("nfeatures", [11, 15])
-@pytest.mark.parametrize("nbackground", [30])
+@pytest.mark.parametrize("n_features", [11, 15])
+@pytest.mark.parametrize("n_background", [30])
 @pytest.mark.parametrize("model", [cuml.SVR])
-def test_kernel_gpu_cpu_shap(dtype, nfeatures, nbackground, model):
-    X, y = cuml.datasets.make_regression(n_samples=nbackground + 5,
-                                         n_features=nfeatures,
-                                         noise=0.1)
-
-    X_train, X_test, y_train, y_test = train_test_split(
-        X, y, test_size=5, random_state=42)
-
-    X_train = X_train.astype(dtype)
-    X_test = X_test.astype(dtype)
-    y_train = y_train.astype(dtype)
-    y_test = y_test.astype(dtype)
+def test_kernel_gpu_cpu_shap(dtype, n_features, n_background, model):
```

Comment:
I was thinking about that, and perhaps it would be good to generate a fixed number of random datasets per session and then many tests can reuse those random datasets, what do you think?